### DISCUSSION TITLE
Prompt Matrix, Prompt compare or x,y,z grid

### DISCUSSION THREAD

**Discussion by Acephalia** [seruva19/kubin#134] [Ideas]
@seruva19 how difficult would it be to implement some kind of function to make a grid out of generated images where one could test out some prompt variations and compare them easily. 

A1111 has this under scripts and it’s super useful to get a better understanding of a models capabilities. Would be great if you could add/suggest something similar. 

I’m happy to help in any way I can.

**Comment by seruva19** [comment]
I've been thinking about this recently, I've been extensively utilizing the feature in A1111 while testing different LoRAs, having this tool for Kandinsky as well would be nice.
Technically, it's not complex, but I need to consider whether it's better to integrate this into the core app or create it as an extension. My primary approach is to move as much functionality as possible to extensions, even if it requires some additional coding.  Regardless, this feature is on my roadmap :)

**Comment by Acephalia** [comment]
The way you’ve structured things I’d say go the extension route. Keeps things nice and tidy. Glad to know it’s on the road map. Don’t hesitate to reach out if you need help with anything.
